x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/lucene/read/SingletonOrdinalsBuilderTests.java

```diff
@@ -53,30 +58,59 @@ public void testReadWithCranky() throws IOException {
             try (IndexReader reader = indexWriter.getReader()) {
                 for (LeafReaderContext ctx : reader.leaves()) {
-                    SortedDocValues docValues = ctx.reader().getSortedDocValues("f");
-                    try (SingletonOrdinalsBuilder builder = new SingletonOrdinalsBuilder(factory, docValues, ctx.reader().numDocs())) {
-                        for (int i = 0; i < ctx.reader().maxDoc(); i++) {
-                            if (ctx.reader().getLiveDocs() == null || ctx.reader().getLiveDocs().get(i)) {
-                                assertThat(docValues.advanceExact(i), equalTo(true));
-                                builder.appendOrd(docValues.ordValue());
+                    int start = 0;
+                    int numDocs = ctx.reader().numDocs();
+                    while (start < numDocs) {
+                        int end = start + randomIntBetween(1, numDocs - start);
+                        BlockLoader.Docs docs = new BlockLoader.Docs() {
+                            @Override
+                            public int count() {
+                                return end;
                             }
-                        }
-                        try (BytesRefBlock build = buildOrdinalsBuilder(builder)) {
-                            for (int i = 0; i < build.getPositionCount(); i++) {
-                                counts.merge(build.getBytesRef(i, new BytesRef()).utf8ToString(), 1, (lhs, rhs) -> lhs + rhs);
+
+                            @Override
+                            public int get(int i) {
+                                return i;
+                            }
+                        };
+                        var columnAtATimeReader = blockLoader.columnAtATimeReader(ctx);
+                        try (BlockLoader.Block block = columnAtATimeReader.read(blockFactory, docs, start)) {
+                            BytesRefBlock result = (BytesRefBlock) block;
+                            BytesRef scratch = new BytesRef();
+                            for (int i = 0; i < result.getPositionCount(); i++) {
+                                counts.merge(result.getBytesRef(i, scratch).utf8ToString(), 1, (lhs, rhs) -> lhs + rhs);
+                            }
+                            boolean singleValue = true;
+                            BytesRef first = BytesRef.deepCopyOf(result.getBytesRef(0, scratch));
+                            for (int i = 1; i < result.getPositionCount(); i++) {
+                                if (first.equals(result.getBytesRef(i, scratch)) == false) {
+                                    singleValue = false;
+                                    break;
+                                }
                             }
+                            BytesRefVector valuesVector = result.asVector();
+                            assertNotNull(valuesVector);
+                            assertThat(valuesVector.isConstant(), equalTo(singleValue));
```

Comment:
If `withIndexSorts` is `true` can we at least guarantee that one block in this test is constant? I'm worried we may not always test that constant block is produced. A dedicated simpler test could maybe guarantee  that?